sql/ddl_internal.sql

```diff
@@ -506,15 +528,23 @@ CREATE OR REPLACE FUNCTION _timescaledb_internal.truncate_hypertable(
 $BODY$
 DECLARE
     hypertable_row _timescaledb_catalog.hypertable;
+    chunk_row _timescaledb_catalog.chunk;
 BEGIN
-
     SELECT * INTO STRICT hypertable_row
     FROM _timescaledb_catalog.hypertable ht
     WHERE ht.schema_name = truncate_hypertable.schema_name
     AND ht.table_name = truncate_hypertable.table_name;
 
-    PERFORM  _timescaledb_internal.drop_chunk(c.id, FALSE)
-    FROM _timescaledb_catalog.chunk c
-    WHERE hypertable_id = hypertable_row.id;
+    FOR chunk_row IN SELECT c.*
+        FROM _timescaledb_catalog.chunk c
+        WHERE hypertable_id = hypertable_row.id 
+        LOOP
+            --TODO: should this cascade?
```

Comment:
Shouldn't this have a cascade_mod the same way as above to match the semantics ? Or do we intend to block any dependencies hypertable -> ordinary tables ?

Reply:
I think it must always cascade, otherwise truncating chunk tables with views or foreign keys referencing them will fail without CASCADE, which it wouldn't do on regular tables. I think creating views or other objects directly on chunks as opposed to the main table should probably come with that caveat.